File: src/main/java/com/blogWebAapp/blogServer/services/PostService.java
package com.blogWebAapp.blogServer.services;

import com.blogWebAapp.blogServer.entities.Post;

import java.util.List;

public interface PostService {
    Post savePost(Post post);

    List<Post> getAllPosts();

    Post getPostById(Long id);

    void likePost(Long id);

    List<Post> searchByName(String name);
}

File: src/app/service/post.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

const BASIC_URL = "http://localhost:8080/";

@Injectable({
  providedIn: 'root'
})
export class PostService {

  constructor(private http: HttpClient) { }

  createNewPost(data: any): Observable<any> {
    return this.http.post(BASIC_URL + 'api/posts', data);
  }
  getAllPosts(): Observable<any> {
    return this.http.get(BASIC_URL + 'api/posts');
  }
  getPostById(postId : number): Observable<any> {
    return this.http.get(BASIC_URL + `api/posts/${postId}`);
  }
  likePost(postId : number): Observable<any> {
    return this.http.put(BASIC_URL + `api/posts/${postId}/like`,{});
  }
  searchByName(name : String): Observable<any> {
    return this.http.get(BASIC_URL + `api/posts/search/${name}`);
  }
}
